unitTests/ui/AnalyticsEndpointTest.ts

```diff
@@ -28,6 +29,31 @@ export function AnalyticsEndpointTest() {
       jasmine.Ajax.uninstall();
     });
 
+    it('exposes a method to resolve URL from a valid SearchEndpoint', () => {
+      const searchEndpoint = new SearchEndpoint({
+        restUri: 'https://platform-eu.cloud.coveo.com/rest/search'
+      });
+      expect(AnalyticsEndpoint.getURLFromSearchEndpoint(searchEndpoint)).toBe('https://platform-eu.cloud.coveo.com/rest/ua');
+    });
+
+    it('exposes a method to resolve URL from an invalid SearchEndpoint', () => {
+      const searchEndpoint = new SearchEndpoint({
+        restUri: 'this-is-not-valid'
+      });
+      expect(AnalyticsEndpoint.getURLFromSearchEndpoint(searchEndpoint)).toBe('this-is-not-valid/rest/ua');
```

Comment:
Related to my comment above on resolving the domain and handling the error case, I guess this test captures both concerns well. I thought the first element in the split array would be `undefined`, but I see here that it is preserving the passed uri.